### ISSUE TITLE
CORS issues

### ISSUE THREAD

**Issue by ethanpil** [dymosoftware/dymo-connect-framework#30]
If I simply run the JavaScript demos untouched I get CORS errors from Chrome. Windows Version 93.0.4577.82 .

**Comment by rfdrake27** [comment]
> DYMO web api accepts all cross origins as default already, and we had to setup new sample website which fixes CORS issue between http://labelwriter.com and https://127.0.0.1:xxxx or https://localhost:xxxx. http://labelwriter.com will be deprecated sooner or later.
> 
> The new sample web pages are found "Sample Html Pages" section in Readme file in https://github.com/dymosoftware/DCD-SDK-Sample.
> 
> The same sample html pages links are: https://s3.amazonaws.com/download.dymo.com/dymo/Software/JavaScript/CheckEnvironment/CheckEnvironment.html https://s3.amazonaws.com/download.dymo.com/dymo/Software/JavaScript/VisitorManagement/VisitorManagement.html https://s3.amazonaws.com/download.dymo.com/dymo/Software/JavaScript/PreviewAndPrintLabel/PreviewAndPrintLabel.html
> 
> Thank you for your patience.

Unfortunately I'm still getting the CORS error even on your sample html pages.

**Comment by cage85** [comment]
@rfdrake27 see above, the CORS issue is probably due to another error occurring before the Dymo Connect Web Service has had a chance to send the correct headers.

**Comment by stormframe** [comment]
Hi,
i had exactly the same Issue.  WIN10H1 aktuell Dymo .82
Afer Reinstalling the Software it was working on Chrome and Edge

**Comment by grezniczek** [comment]
@dymosoftware it would be really nic if the webservice would handle internal errors properly, i.e. send correct headers from the error handler.

It's really about time this was implemented.

**Comment by akash-s-simformsolutions** [comment]
> ![image](https://user-images.githubusercontent.com/76422838/145372800-e15f4a70-45ba-4724-b839-b5e8b8cea22a.png)
> 
> I am loading label and update label text fields and printing label using latest dymo connect software and windows10 but i am keep getting CORS issue.
> 
> tried with Chrome://flags setting but not working.
> 
> can anybody please help

@ethanpil @stormframe @rfdrake27 @ande2407 Please let me know if any anybody has working solution or any suggestions about above issue.

@dymosoftware ??

**Comment by cage85** [comment]
@akash-s-simformsolutions the PrintLabel endpoint is throwing a 500. Maybe your (or Dymos) test page is using a metric-based label type and your Windows installation is non-metric (e.g. US English) or vice-versa.

**Comment by ErnstWolthaus** [comment]
Hi all,

I've managed to get the Dymo LabelWriter 450 working from Dynamics 365 CE.
It took me a while and I thought I also had an issue with CORS.

I debugged the JS lib of Dymo and in the end it turned out to be an error in my label xml. So, the CORS error message might be a 'red herring', meaning if your sending the printer invalid data, you might get the wrong error message...
(I think @cage85 also is pointing to this...)

Now my label xml is OK and I have no issues using the Dymo Labelwriter 450. So please make sure your data is correct.
(of course, you do need to have the Dymo service running on your local computer)

Good luck!
Ernst

**Comment by akash-s-simformsolutions** [comment]
@cage85  Thanks for sharing details. i will check about label type.

@ErnstWolthaus Thanks for sharing details. I am also trying same with Dymo label Printer 450 4XL. 
I have created Label using Latest Dymo connect software and i am using that label and updating few dynamic values.
```
var labelXml = "XML file content geneated by software .dymo file content"
var label = dymo.label.framework.openLabelXml(labelXml);

// Updating some values in label to update it dynamically
label.setObjectText("ITextObject0", "XYZ");
label.setObjectText("ITextObject4", "ABC");

// After updating label i check if its valid or not
label.isValidLabel();  <-- This prints True

// Print label 
label.print("DYMO LabelWriter 4XL");   <--- This is where getting errors.
```
Please check above flow and let me know anything if i am missing. Thanks in advance.

@ErnstWolthaus : If you can share how you created label that would be also helpful.

**Comment by ErnstWolthaus** [comment]
Hi @akash-s-simformsolutions

1. the labelXml is not correct... it's not xml, so dymo.label.framework.openLabelXml will not process that.
2. are you sure the printername is correct? 

If you are a developer, you can download the full version of the framework and debug into the library to see what's wrong...

I have used this code (changed the labelXml to actual xml, see attachment), and it works for me.
FYI: I have created the label xml with Dymo Connect software. Just create the label and save it to disk. Then use the xml content of that file.

[labelsample.txt](https://github.com/dymosoftware/dymo-connect-framework/files/7743726/labelsample.txt)

**Comment by akash-s-simformsolutions** [comment]
Hi @ErnstWolthaus Thanks for quick response and example.

**Comment by timint** [comment]
I am too having this CORS issue and can't get around it.
This is on both MacOS and Windows with the latest JS framework on Github.
Firefox, Chrome and Edge are all affected.

Isn't this a problem with stricter security policies being rolled out from the browser makers when no CORS policy is defined?

> VM12:1 Access to XMLHttpRequest at 'https://127.0.0.1:41951/DYMO/DLS/Printing/PrintLabel' from origin 'https://www.trionic.se' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

The following did not solve my problem:

> chrome://flags/#block-insecure-private-network-requests
> disabled

I figured out why StatusConnected Works
https://127.0.0.1:41951/DYMO/DLS/Printing/StatusConnected
> Returns true
![image](https://user-images.githubusercontent.com/359192/149422925-fb5b1c57-2762-43c5-bc2c-8aeb99ad987e.png)

And PrintLabel doesn't:
![image](https://user-images.githubusercontent.com/359192/149422827-c2aa7896-3a92-4fc2-af8b-490da8c3aa76.png)

DYMO needs to consistently add the CORS policy to all server responses, not just some.

**Comment by 0xE1** [comment]
We have same issue as well, the only workaround we managed to work is to use DYMO Label Software 8.5.4 and block DYMO Connect Service from using first port, then it works without issues.

I will see if I can connect debugger to DYMO service to see what it fails on...

**Comment by timint** [comment]
@0xE1 They are missing the Access-Control-Allow-Origin: * header in the HTTP response for PrintLabel.

**Comment by 0xE1** [comment]
For every response? Was mentioned "not just some", so it might be that the service just throws an exception for most requests and then response does not include necessary headers.

**Comment by timint** [comment]
@0xE1 If any DYMO Connect response does not hold an Access-Control-Allow-Origin HTTP header in the response, the browser will return a HTTP 500 error. They implemented it on some responses like /StatusConnected, and /GetPrinters but not /PrintLabel.

**Comment by dymosoftware** [comment]
As @ande2407 wrote in previous comments, have you tried to include the script in your web project?
https://s3.amazonaws.com/download.dymo.com/dymo/Software/JavaScript/dymo.connect.framework.js

**Comment by timint** [comment]
@dymosoftware This issue ticket is already addressed not to be a user problem. But a problem in DYMO.WebApi.Win.Host.exe.
https://github.com/dymosoftware/dymo-connect-framework/issues/30#issuecomment-1012600893

**Comment by dymosoftware** [comment]
@timint We are beginning to investigate this issue. Thanks for your help so far diagnosing it.

**Comment by timint** [comment]
@dymosoftware I am very delighted to hear this. So are my clients. 👍 🙂

**Comment by xd-luke** [comment]
@dymosoftware it has been over a week. How long does it take your developers to add this one header? Our project is currently stalled because of this issue. Can you please provide an accurate time estimate? We need to make a decision to downgrade versions to get this working in our systems. We would prefer not to downgrade for obvious reasons. If this will be fixed soon we will wait. But, if not, we have not other choice but to downgrade.

**Comment by andythedude** [comment]
> 

I don't think it's a high priority for them. Seems as if all their focus is on the actual printers and the printer desktop software, whereas a library like this is an afterthought. It took them almost 3 months to even start working on this so i wouldn't count on them fixing it with any degree of urgency.

**Comment by dymosoftware** [comment]
Hi all, could you try to run your app with this preview version of [DLSSidecar_1.4.3-147-beta-WS.exe](https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Windows/DLSSidecar_1.4.3-147-beta-WS.exe) installed?

Let us know if you still have issues with API headers.

**Comment by timint** [comment]
@dymosoftware The CORS problem is fixed. But a new fatal error was returned.

> Error printing label. Exception: 'Element' is an invalid XmlNodeType. Line 22, position 4.

I tried the XML of two different .dymo files. Same line error.

![image](https://user-images.githubusercontent.com/359192/150658242-f0b757b2-e970-47a2-a54e-19bdd5f1b1e6.png)

**Comment by dymosoftware** [comment]
> @dymosoftware The CORS problem is fixed. But a new fatal error HTTP 400 was returned.
> 
> > Error printing label. Exception: 'Element' is an invalid XmlNodeType. Line 22, position 4.
> 
> I tried the XML of two different .dymo files. Same line error. (Yes I removed the Byte Order Mark.. that's another confirmed bug see issue #45 ).
> 
> ![image](https://user-images.githubusercontent.com/359192/150658242-f0b757b2-e970-47a2-a54e-19bdd5f1b1e6.png)
> 
> ![image](https://user-images.githubusercontent.com/359192/150658324-af0eb4e8-a251-4f76-b0c1-b76410cbd8cb.png)

@timint could you attach the payload sent through the PrintLabel request? Specially the **labelXML** value.

**Comment by timint** [comment]
@dymosoftware I have created a separate issue ticket for that. See issue #47

**Comment by guy-shahine** [comment]
@dymosoftware is there a plan to fix the DYMO CONNECT version 1.4.2 on macOS? We're facing an error on macOS Big Sur `DYMO Label Plugin is not installed.` Thank you
![Screen Shot 2022-01-25 at 21 14 23](https://user-images.githubusercontent.com/1103409/151102990-cdac0a12-bc3e-4722-8281-03ba702add4f.png)

**Comment by a1development** [comment]
> Hi all, could you try to run your app with this preview version of [DLSSidecar_1.4.3-147-beta-WS.exe](https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Windows/DLSSidecar_1.4.3-147-beta-WS.exe) installed?
> 
> Let us know if you still have issues with API headers.

OK with the Beta installed ( and rebooted ) it works fine! Thank you!
Do you know at which Version this fix will be included?

**Comment by xd-luke** [comment]
DLSSidecar_1.4.3-147-beta-WS.exe Looks to be working for me as well. Thank you!

**Comment by dymosoftware** [comment]
> > Hi all, could you try to run your app with this preview version of [DLSSidecar_1.4.3-147-beta-WS.exe](https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Windows/DLSSidecar_1.4.3-147-beta-WS.exe) installed?
> > Let us know if you still have issues with API headers.
> 
> OK with the Beta installed ( and rebooted ) it works fine! Thank you! Do you know at which Version this fix will be included?

Fix will be included in 1.4.3 release, expected on March.

**Comment by akash-s-simformsolutions** [comment]
@ethanpil : You can find exe and mac application on this URL [Beta Version](https://github.com/dymosoftware/Beta-Version/tree/main/DYMO%20Connect%20Windows)

**Comment by dymosoftware** [comment]
> I'm the originator of this issue and I just came back happy to find this closer to being resolved, however it seems the link to DLSSidecar_1.4.3-147-beta-WS.exe is now invalid?

You can find DLSSidecar_1.4.3-150-beta-WS [here](https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Windows/DLSSidecar_1.4.3-150-beta-WS.exe)

**Comment by marvinvdongen** [comment]
@dymosoftware it seems that link is not working anymore. It mentions something about LFS quota being reached. Can you please put the EXE somewhere we can download it?

**Comment by grezniczek** [comment]
Yep. See https://github.com/dymosoftware/Beta-Version/issues/1

**Comment by marvinvdongen** [comment]
> > I'm the originator of this issue and I just came back happy to find this closer to being resolved, however it seems the link to DLSSidecar_1.4.3-147-beta-WS.exe is now invalid?
> 
> @ethanpil You can find DLSSidecar_1.4.3-150-beta-WS [here](https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Windows/DLSSidecar_1.4.3-150-beta-WS.exe)

I installed this version on one of our client machines (Win7) . It doesnt work at all. The Dymo Connect GUI crashes with this info:

> Problem signature:
  Problem Event Name:	CLR20r3
  Problem Signature 01:	DYMOConnect.exe
  Problem Signature 02:	1.4.3.150
  Problem Signature 03:	61f4072b
  Problem Signature 04:	mscorlib
  Problem Signature 05:	4.6.1590.0
  Problem Signature 06:	5787ed44
  Problem Signature 07:	4168
  Problem Signature 08:	0
  Problem Signature 09:	System.Windows.Markup.XamlParse
  OS Version:	6.1.7601.2.1.0.320.65
  Locale ID:	1043
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

And the Connect services crashes too:
![image](https://user-images.githubusercontent.com/28144671/152231827-ebd5b19e-9dd9-4616-8dff-4d747a6f2df9.png)

**Comment by marvinvdongen** [comment]
On another PC with Win7 it does install succesfully. The CORS issue still occurs though:

`mainjs?v=JW3WR4tR-tptGWtItcMMRRypi85OXD4RYaO3umL1AOI1:1 Access to XMLHttpRequest at 'https://127.0.0.1:41951/DYMO/DLS/Printing/GetPrinters' from origin 'XXXXXXXX' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by marvinvdongen** [comment]
> On another PC with Win7 it does install succesfully. The CORS issue still occurs though:
> 
> `mainjs?v=JW3WR4tR-tptGWtItcMMRRypi85OXD4RYaO3umL1AOI1:1 Access to XMLHttpRequest at 'https://127.0.0.1:41951/DYMO/DLS/Printing/GetPrinters' from origin 'XXXXXXXX' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`
> 
> ![image](https://user-images.githubusercontent.com/28144671/152237207-0f4bfdb1-bf7b-4c75-90c2-651e76cb5e6a.png)
> 
> Seems like the allow-origin is still missing here.

Is there any news on this? We have been forced to revert back to Chrome 97 (Chromium) for many (100+) of our clients. Which is far from ideal and not future-proof at all.

**Comment by pbhogan** [comment]
What's the status on Mac? The lack of any kind of documentation makes this all very confusing, but from what I can tell DYMO Connect doesn't have a web service on Mac, so we're running Dymo Label Service. As of recent Chrome versions it fails CORS checks like everyone is pointing out, but it also completely hangs on certain requests, which, I believe, is due to a CORS preflight request the Chrome issues. The web service takes the connection but never responds, so it just hangs indefinitely.

You can reproduce this with cURL as follows:

```
curl -H "Origin: https://example.com" -H "Access-Control-Request-Method: GET" -H "Access-Control-Request-Headers: X-Requested-With" -X OPTIONS --verbose https://127.0.0.1:41951/DYMO/DLS/Printing/GetPrinters
```

Firefox doesn't seem to have this issue (yet) but it's probably only a matter of time before printing with the web service is broken there too.

**Comment by 0xE1** [comment]
Stable DCD on Mac is missing Web Service, Beta version has it - but it fails to start properly due to SSL error.

We have to ask people to use `--disable-features=PrivateNetworkAccessRespectPreflightResults` flag to be able to use DYMO on Chrome 98 since it does not handle prefligh requests in any of expected ways, it simply keeps connection opened indefinitely and 3000ms timeout on request actually never times out.

**Comment by ashleybot** [comment]
Is there documentation for the api endpoints? It seems that would help in narrowing down some issues.

I've read through the entire thread. These are things that don't work:
❌  Printing from DYMO's sample webpages on amazon
❌  Printing from @ande2407 's example webpage
❌  Printing from my own remote web page (yes, with the dymo connect js library directly within my codebase and not using a cdn or external reference)
❌  Disabling `Block insecure private network requests` does not fix the issue for a remote requests to localhost

✅  The only fix for me is to run the web app on localhost (which isn't a fix because that's my local dev environment and not available to my users). 
✅  I have no issues with my label's xml and no HTTP 500 errors. Everything prints fine from local. I use the exact same label structure and dymo js in code deployed to other servers, so no change there. Additionally, I get the error on any call to the printer webapi including just a status check `/DYMO/DLS/Printing/StatusConnected`

**If we could get some documentation to understand how to form a good request then we could make our own calls to the endpoint. So, if it is Chrome hiding a 500 error for whatever reason at least we could diagnose properly.**

The CORS header must the issue because it just doesn't happen when the code is running from localhost. Wouldn't we see issues on local if there were in fact mal-formed label xml or otherwise? 

`Access to XMLHttpRequest at 'https://localhost:41951/DYMO/DLS/Printing/StatusConnected' from origin '{REMOTE}' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by mkvist** [comment]
DYMO just uploaded (4 hours ago) a new version of the DLSSidecar and it works for me. It is version 1.4.3.123-beta. 

Yesterday i tried with the DLSSidecar_1.4.3-150-beta-WS.exe and it did not work. I had to uninstall Dymo Connect, before i was able to install the 1.4.3.123 version. 

https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Windows/DLSSidecar-1.4.3.123-beta.exe

**Comment by henrik** [comment]
Direct link to the corresponding (?) Mac app, uploaded at the same time: https://github.com/dymosoftware/Beta-Version/blob/main/DYMO%20Connect%20Mac/DYMOConnect-1.4.3.93-beta.pkg

Not in a position to try it out currently.

**Comment by miguelduarte42** [comment]
I get the following error on 1.4.3.93-beta

![Screenshot 2022-02-15 at 16 27 53](https://user-images.githubusercontent.com/1619310/154105098-e11d6e29-5a70-4cad-aec4-b84a123b8020.png)

**Comment by dymosoftware** [comment]
There is an work-around for the beta version Web Api issue.
https://github.com/dymosoftware/Beta-Version/issues/2

**Comment by ashleybot** [comment]
> There is an work-around for the beta version Web Api issue. [dymosoftware/Beta-Version#2](https://github.com/dymosoftware/Beta-Version/issues/2)

Following @blester-mvc's instructions to remove everything and start from a blank slate worked for me.

**Comment by pbhogan** [comment]
Well, the CORS issue has now hit with the latest Firefox update, so the workaround we had of using Firefox instead of Chrome is now dead. 

When is this issue going to get resolved? It's been months now!

Running DYMO Connect 1.4.3.103 for Mac

This is in the Firefox console:

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://127.0.0.1:41951/DYMO/DLS/Printing/StatusConnected. (Reason: CORS request did not succeed). Status code: (null).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://localhost:41951/DYMO/DLS/Printing/StatusConnected. (Reason: CORS request did not succeed). Status code: (null).

**Comment by zsoden** [comment]
I implemented the Dymo.connect.framework on an in-house PHP/Javascript system in late 2021 and it's been working fine until recently - I presume due to the Firefox updates. We don't use Chrome on any computers. 

We too are now getting the CORS error: "Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://127.0.0.1:41951/DYMO/DLS/Printing/StatusConnected. (Reason: CORS request did not succeed). Status code: (null)."

The networking tab in Firefox inspector shows no response, but I presume that's because the failed CORS check completely blocks the response.

I tried to make a copy of the framework with https disabled so that I could do some testing in plaintext (and potentially see error information) but I don't have enough knowledge to do that. The dymo.connect.framework.js file is far more complex than anticipated.

**Comment by timint** [comment]
@dymosoftware This issue has been ongoing for a year soon. We know what he cause of the problem is. We know how to properly fix it. Why can't we have a proper fix?

This is a total blockstopper for my clients who run the printer.

Simply implement CORS headers for all your Web Service responses and we should be good.

`Access-Control-Allow-Origin: *`

**Comment by lmason98** [comment]
The recently updated the beta version for windows of the dymo connect software. I was running into this issue and installing that version cleared it up for me. https://github.com/dymosoftware/Beta-Version

**Comment by timint** [comment]
@lmason98 What if they could spend some time on actually maintaining these issue reports? Nah prolly takes a rocket engineer for that.

**Comment by ethanpil** [comment]
I bought one of these Brother printers instead. 
Their JS SDK works great and the machine is a beast!
https://amzn.to/3YBXum6

**Comment by zsoden** [comment]
I too gave up and bought some used Zebra printers. I had it printing our stuff in ZPL in just a couple of hours, vs days of initial time investment getting Dymo Framework to run. So simple to use, both using the Zebra print service or printing directly to the printer's IP address!

**Comment by timint** [comment]
@zsoden That's the right thing to do. I tried reaching out to DYMO management regarding this complete disaster they have for development. But they didn't want to hear from me. Totally their loss. I have discontinued all DYMO work. Some of my clients also got the Zebra instead. Happy to hear the move went well. I will spread the word.

**Comment by dartacus** [comment]
Wow, you don't see this in Dymo marketing, do you? Nothing except a few obtuse replies from the devs, never resolved, OP has had to buy alternative hardware.

I have exactly the same issue with the latest version of Dymo Connect, so it's clear that Dymo aren't going to fix this.

I would point anyone in the same boat in the direction of QZTray, which can be configured to print to a Dymo label printer from your web app, securely.

And if you're currently making decisions as to what to spend your label printer budget on, well - I think we can all draw our own conclusions from this absolute disaster of a support experience.

**Comment by innerpurpose** [comment]
This has been devastating for me as founder personally doing tech support. Thank you to anyone helping give us solutions.

**Comment by dartacus** [comment]
> This has been devastating for me as founder personally doing tech support. Thank you to anyone helping give us solutions.

The previous wording of my comment at the time was probably coloured by several wasted hours grappling with this issue, so I apologise for that: it was needlessly pejorative but I was not in a great frame of mind. I hope you do find a solution: the older version of the software functioned perfectly for many years printing direct from a browser (still is, on one PC).

**Comment by timint** [comment]
@dartacus Don't feel like you need to apologize. We have all been there. This repository, the development team, and upper management has been a total disaster. Still is. Fact. I say speaking our minds is the best we can do to make ourselves heard. One day they might care.

**Comment by craftyweb** [comment]
Hi guys!

I'm also having trouble with the CORS issue for a while now. I'm trying to print labels from an ecommerce website so that my client can print the labels when they added new products.

Were now looking for other alternatives than Dymo. Any good suggestions? I prefer a printer that I can use with an javascript SDK

Thanks!

**Comment by 0xE1** [comment]
At this stage it's honestly better to create label pdf either client-side or server-side and present it for printing using normal printing dialog (for example using `window.print()` of a child frame or window),
and wait for recently proposed [Web Printing API](https://github.com/WICG/proposals/issues/120) to be implemented so you can engage any system printer directly.

I have already created a DYMO label parser (pre-Connect format), preview renderer and PDF export that works perfectly well with [chrome.printing](https://developer.chrome.com/docs/extensions/reference/printing/) API, I just need to clear it with company before I can publish it on Github. Ideally creating vendor-indepentend label format would be best output so it can be printed on anything while maintaining element placement if different paper size is used, though even current form works on any printer.

**Comment by luckyexpert** [comment]
For anybody else facing this issue, uninstalling and reinstalling Dymo Connect solved my issue - the CORS policy errors went away and I was able to debug again.

**Comment by Muschke** [comment]
I was on windows 10. I saw about (KB5015807) on the software download center. I saw this one inside my update but not the fixed. So I installed all possible updates.  Also I made sure I took the correct compatible software for the printer. https://www.dymo.com/support?cfid=dymo-compatibility-chart
Than I rebooted and everything seems to be okay now.

**Comment by AzharAbbas10** [comment]
Hi everyone,
I'm working on project where my client want to print label direct from website it work fine 3 4 times then it is giving me error of CORS anyone have any idea how can I handle this error?
TIA

**Comment by AzharAbbas10** [comment]
> For anybody else facing this issue, uninstalling and reinstalling Dymo Connect solved my issue - the CORS policy errors went away and I was able to debug again.

hi

**Comment by dartacus** [comment]
I had a certain amount of success using QZTray to print to a dymo - it's fine as long as your formatting requirements aren't too complex. But my main solution is still just keeping an old copy of the dymo software for workstations which do labels.

**Comment by Werner-Calian** [comment]
A brief dig through the framework js file shows that the following endpoints are being hit:

/DYMO/DLS/Printing/StatusConnected
/DYMO/DLS/Printing/GetPrinters
/DYMO/DLS/Printing/OpenLabelFile
/DYMO/DLS/Printing/PrintLabel
/DYMO/DLS/Printing/PrintLabel2
/DYMO/DLS/Printing/RenderLabel
/DYMO/DLS/Printing/LoadImageAsPngBase64
/DYMO/DLS/Printing/GetJobStatus
/DYMO/DLS/Printing/Is550Printer
/DYMO/DLS/Printing/GetConsumableInfoIn550Printer

Adding the vlue below to your  Content-Security-Policy uner "content-src 'self'"  should sort out the CORS issue:
https://127.0.0.1:*/DYMO/DLS/Printing/StatusConnected https://127.0.0.1:*/DYMO/DLS/Printing/GetPrinters https://127.0.0.1:*/DYMO/DLS/Printing/OpenLabelFile https://127.0.0.1:*/DYMO/DLS/Printing/PrintLabel https://127.0.0.1:*/DYMO/DLS/Printing/PrintLabel2 https://127.0.0.1:*/DYMO/DLS/Printing/RenderLabel https://127.0.0.1:*/DYMO/DLS/Printing/LoadImageAsPngBase64 https://127.0.0.1:*/DYMO/DLS/Printing/GetJobStatus https://127.0.0.1:*/DYMO/DLS/Printing/Is550Printer https://127.0.0.1:*/DYMO/DLS/Printing/GetConsumableInfoIn550Printer

**Comment by luckyexpert** [comment]
Its October 2024 and this is still an issue. The only solution I can find is uninstall and reinstall Dymo connect. My customers are tired of it and I am tired of it so first chance I get I will be integrating with Zebra and pushing all my customers that way. Seems this is  trivial fix so not sure why Dymo is ignoring this

**Comment by timint** [comment]
I wrote a letter to Dymo trying to reach management, saying I had critical information about their software team that should be in their best interest. They replied they will not arrange for somone to talk to. Instead, I should contact 1st line support. At this point they don't deserve customers.

**Comment by luckyexpert** [comment]
If Dymo is paying attention to these comments (doesn't seem like it), I think there is more to this issue. I installed a CORS plugin on chrome to try and find a workaround for this issue for my customers. I was able to get Chrome to look past the CORS issue, however the console in chrome just showed a 500 error on the attempt to call the local dymo service. I used Postman to try and call the same URL (https://127.0.0.1:41951/DYMO/DLS/Printing/GetPrinters) and it also returned a 500 error. I installed an update the dymo connect and all of a sudden it all started working again. in the past this as been the case where if Dymo connect is reinstalled, the service starts work again. Sure enough, calling the URL above in postman returns the XML of the installed printers rather than the 500 error. So I think somehow the service is crashing which is causing this issue, not necessarily a CORS issue that it looks like

**Comment by ChristianAlteri** [comment]
how has this not been resolved lol. Im having the issue. Ripping my hair out then came here and so is everyone else

**Comment by timint** [comment]
Poor tech and crew, and awful management. For the sake of all Dymo users I hope this gets turned around. There are plenty of good kids and repo owners around on Github who can do an awesome job if Dymo could finance their time.

**Comment by ChristianAlteri** [comment]
Completely agree!
